docs/guides/prisma.md

```diff
@@ -0,0 +1,360 @@
++model User {
++  id    Int     @id @default(autoincrement())
++  email String  @unique
++  name  String?
++  posts Post[]
++}
+
+model Post {
+  id        Int      @id @default(autoincrement())
+  createdAt DateTime @default(now())
+  title     String
+  content   String?
+  published Boolean  @default(false)
+  viewCount Int      @default(0)
++  author    User?    @relation(fields: [authorId], references: [id])
++  authorId  Int?
+}
+
+model versions {
+  version BigInt    @id
+  name    String?
+  md5     String?
+  run_at  DateTime? @db.Timestamptz(6)
+
+  @@ignore
+}
+```
+
+Next, use the `@ruheni/db-diff` CLI tool to generate `up` and `down` migrations:
+
+```bash
+npx db-diff
+```
+
+This command will generate up and down migrations based off of your Prisma schema. If you are currently using a different path to store the migration, you can provide the `--migrations-dir` flag followed by the path.
+
+Next, apply the generated migration using the Platformatic CLI:
+
+```bash
+npx platformatic db migrations apply
+```
+
+And you're done!
+
+## Extending functionality with Plugins using Prisma Client
+
+:::danger
+The following functionality is only supported on Node v18 
+:::
+
+You can use Prisma Client to 
+```bash
+npx prisma generate
+```
+
+The above command will generate Prisma Client based off your Prisma schema.
+
+```js
+// ./plugin.js
+
+// 1. 
+const { PrismaClient } = require("@prisma/client")
+
+// 2
+const prisma = new PrismaClient()
+
+module.exports = async (app) => {
+  app.log.info('plugin loaded')
+
+  //3.
+  app.graphql.extendSchema(`
+    extend type Mutation {
+      incrementPostViewCount(id: ID): Post
+    }
+  `)
+
+  //4.
+  app.graphql.defineResolvers({
+    Mutation: {
+      incrementPostViewCount: async (_, { id }) => {
+        // 5. 
+        const post = await prisma.post.update({
+          where: {
+            id: Number(id)
+          },
+          data: {
+            viewCount: {
+              increment: 1
+            }
+          }
+        })
+
+        // 6.
+        if (!post) throw new Error(`Post with id:${id} was not found`)
+        // 7.
+        return post
+      }
+    }
+  })
+}
+```
+
+The previous snippet does the following:
+
+1. Imports Prisma Client.
+1. Creates a new instance of Prisma Client.
+1. Defines the `incrementPostViewCount` mutation type definition. It also defines the corresponding argument and the return type.
+1. Defines the resolver for your mutation.
+1. Makes a query to the database on the Post model to increment a post's view count.
+1. Adds error handling in the event a record with the matching id was not found.
+1. Returns the updated post on success.
+
+Once you start your server, the query should be included in your GraphQL schema.
+
+You can also use Prisma Client to define REST API endpoints.
+
+Alternatively, you can use the [`@sabinthedev/fastify-prisma`](https://github.com/sabinadams/fastify-prisma) plugin to query your database using Prisma Client. 
```

Comment:
The `@sabinthedev/fastify-prisma` plugin takes care of correctly shutting down the database connection and can be used to make the Prisma Client available in all Fastify plugins that are loaded in the application.

Are there any benefits to importing `PrismaClient` and creating a client instance vs using the plugin? If not, I would suggest integrating the steps for using plugin in to the previous steps and removing the direct usage of `PrismaClient` above.

Reply:
The Plugin currently uses Prisma v4.2.0. I've created a PR to make Prisma Client a peer dependency instead of a dependency. Once the PR is merged, I can update this section.

Reply:
Does the mismatch between the 4.x minor versions of `@prisma/client` cause any issue?

Reply:
For the purposes of this guide, I'm not sure that's an issue. Would you be happy to update this section anyway?